fabric-client-gametest-api-v1/src/client/java/net/fabricmc/fabric/api/client/gametest/v1/ClientGameTestInput.java

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.gametest.v1;
+
+import java.util.function.Function;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.option.GameOptions;
+import net.minecraft.client.option.KeyBinding;
+import net.minecraft.client.util.InputUtil;
+
+/**
+ * The client gametest input handler used to simulate inputs to the client.
+ */
+@ApiStatus.NonExtendable
+public interface ClientGameTestInput {
```

Comment:
I wonder if it'd be more useful if those APIs used AutoCloseable, so that it can be wrapped in try-with-res.

Reply:
This is planned for test worlds (and already implemented for test servers), but why would the input need closing?

Reply:
So that you don't forget depressing?

Reply:
I don't think that's a great reason to overcomplicate the API. Input is always valid to use for the whole duration of the test. All keys are automatically unpressed after every test anyway so it won't affect the other tests.